frontend/__tests__/components/layout/Header.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { fireEvent, render, waitFor } from "@testing-library/react"
+import Header from "../../../src/components/layout/Header"
+import { UserState } from "../../../src/types/Login"
+import { UserContext, UserDispatchContext } from "src/context/user-info"
+import { translationMock } from "../../../jest.setup"
+
+describe("Header tests", () => {
+  it("User logs out successfully", async () => {
+    const dispatchMock = jest.fn()
+    const userState: UserState = {
+      loading: false,
+      info: {
+        auths: {
+          github: {
+            avatar: "https://avatars.githubusercontent.com/u/27268838?s=200&v=4",
+            login: "devflat",
+          },
+          gitlab: undefined,
+          gnome: undefined,
+          kde: undefined,
+        },
+        "is-moderator": false,
+        "dev-flatpaks": [],
+        "owned-flatpaks": [],
+        displayname: "dev-flatpak",
+        "accepted-publisher-agreement-at": ""
+      },
+    }
+
+    translationMock.mockImplementation((args) => {
+      if (args === "user-avatar") {
+        return `${userState.info.displayname}'s avatar`
+      }
+      return args
+    })
+
+    const expectedUrlValue = "https://wiki.gnome.org"
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve({ url: expectedUrlValue }),
+      }),
+    ) as jest.Mock
+
+    const { getByText, container } = render(
+      <>
+        <UserContext.Provider value={userState}>
+          <UserDispatchContext.Provider value={dispatchMock}>
+            <Header />
+          </UserDispatchContext.Provider>
+        </UserContext.Provider>
+      </>,
+    )
+
+    const userIconBeforeLogout = container.querySelector("img.rounded-full")
```

Comment:
I would much rather see us using testing library here to also ensure accessability

https://testing-library.com/docs/queries/about#priority

Reply:
changed to getByRole